x/username/model.go

```diff
@@ -0,0 +1,80 @@
+package username
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+)
+
+func init() {
+	migration.MustRegister(1, &Token{}, migration.NoModification)
+}
+
+func (ba *BlockchainAddress) Validate() error {
+	switch n := len(ba.BlockchainID); {
+	case n < 3:
+		return errors.Wrap(errors.ErrInput, "blockchain ID too short")
```

Comment:
Nice to give special errors for length.
Can you also assert the values? To avoid emojis and other such nonsense

https://github.com/iov-one/weave/blob/master/cmd/bnsd/x/nft/username/model.go#L17
`var validBlockchainID = regexp.MustCompile(`^[a-zA-Z0-9_.-]{4,32}$`).Match`

Reply:
Addressed in https://github.com/iov-one/weave/pull/771/commits/9eef83f3027589acba690e2f500002722ef21cca